src/ring_theory/unique_factorization_domain.lean

```diff
@@ -955,7 +968,28 @@ begin
   rwa [prod_factors, prod_factors] at this
 end
 
-include dec dec'
+include dec dec' dec_irr
+
+theorem eq_factors_of_eq_counts [nontrivial α] {a b : associates α} (ha : a ≠ 0) (hb : b ≠ 0)
+  (h :  ∀ (p : associates α) (hp : irreducible p), p.count a.factors = p.count b.factors) :
+  a.factors = b.factors :=
+begin
```

Comment:
I think this also holds without `ha` and `hb`? From my quick reading, `count a.factors = λ x, 0 ↔ a =0`.

Reply:
I don't think this is an iff; if a is a unit, then we also get `count a.factors = λ x, 0` (but in that case `a.factors` is empty, while `0.factors` is `none`). That's why I included `ha` and `hb`.

Reply:
oops! of course, that makes sense.